crates/outbound-mysql/src/lib.rs

```diff
@@ -244,11 +245,89 @@ impl OutboundMysql {
 async fn build_conn(address: &str) -> Result<mysql_async::Conn, mysql_async::Error> {
     tracing::log::debug!("Build new connection: {}", address);
 
-    let connection_pool = mysql_async::Pool::new(address);
+    let opts = build_opts(address)?;
+
+    let connection_pool = mysql_async::Pool::new(opts);
 
     connection_pool.get_conn().await
 }
 
+/// The mysql_async crate does not currently support SSL opts properly. This function is a workaround to
+/// manually set SSL if opts if the use requests them. We only support ssl-mode in the query as per
+/// https://dev.mysql.com/doc/connector-j/8.0/en/connector-j-connp-props-security.html#cj-conn-prop_sslMode.
+fn build_opts(address: &str) -> Result<Opts, mysql_async::Error> {
+    let url = Url::parse(address)?;
+
+    let is_ssl_param = |s: &str| ["ssl-mode", "sslmode"].contains(&s.to_lowercase().as_str());
+
+    let use_ssl = url
+        .query_pairs()
+        .any(|(k, v)| is_ssl_param(&k) && v.to_lowercase() != "disabled");
+
+    let query_without_ssl: Vec<(_, _)> = url
+        .query_pairs()
+        .filter(|(k, _v)| !is_ssl_param(k))
+        .collect();
+    let mut cleaned_url = url.clone();
+    cleaned_url.set_query(None);
+    for pair in query_without_ssl {
+        cleaned_url
+            .query_pairs_mut()
+            .append_pair(&pair.0.to_string()[..], &pair.1.to_string()[..]);
+    }
+
+    Ok(OptsBuilder::from_opts(cleaned_url.as_str())
```

Comment:
This may be a stupid question but... do we need to do the `cleaned_url` thing?  Could we not Do the `from_opts` with the original URL, and then call `ssl_opts` to override whatever incorrect SSL option was inferred from that original URL?

Reply:
Valid question, this is how the fix originally worked when I was debugging with Joel. However, the `mysql_async` library blows up if you pass it a query parameter with `sslMode` in it. This is why I have to clean it out of the url. Hopefully they can fix this in upstream when I file the issue.

Reply:
Ah, then maybe clarify the comment to say you _can't even pass_ a SSL parameter without it blowing up, more than "doesn't currently support properly" which I took to mean "didn't recognise the parameter / recognised it wrong."  A link to the upstream issue would be really good too, so that a maintainer can easily check "is this fixed yet."

Reply:
Will do.